mava/components/jax/building/environments.py

```diff
@@ -24,11 +24,48 @@
 from mava.components.jax import Component
 from mava.core_jax import SystemBuilder
 from mava.environment_loop import ParallelEnvironmentLoop
+from mava.utils.sort_utils import sort_str_num
 
 
 @dataclass
-class ExecutorEnvironmentLoopConfig:
+class EnvironmentSpecConfig:
     environment_factory: Optional[Callable[[bool], acme.core.Worker]] = None
+
+
+class EnvironmentSpec(Component):
+    def __init__(self, config: EnvironmentSpecConfig = EnvironmentSpecConfig()):
+        """[summary]"""
+        self.config = config
+
+    def on_building_init_start(self, builder: SystemBuilder) -> None:
+        """[summary]"""
+
+        builder.store.environment_spec = specs.MAEnvironmentSpec(
+            self.config.environment_factory()
+        )
+
+        builder.store.agents = sort_str_num(
+            builder.store.environment_spec.get_agent_ids()
+        )
+        builder.store.extras_spec = {}
+
+    @staticmethod
+    def name() -> str:
+        """_summary_"""
+        return "environment_spec"
+
+    @staticmethod
+    def config_class() -> Optional[Callable]:
+        """Config class used for component.
+
+        Returns:
+            config class/dataclass for component.
+        """
+        return EnvironmentSpecConfig
+
+
+@dataclass
+class ExecutorEnvironmentLoopConfig:
```

Comment:
What was the reason for creating two components (EvironmentSpec and EnvironmentLoop) instead of one?

Reply:
@DriesSmit From my perspective it makes sense to split the two. `EvironmentSpec` is only about the spec of the env, while  `EnvironmentLoop` is env loop that gets run e.g. `ParallelEnvironmentLoop`. They serve different uses from my perspective and it also make testing easier, e.g. a component might only need an `EvironmentSpec` to be tested and not nec a whole env loop. Thoughts?

Reply:
If it makes testing easier it sounds fine to me :slightly_smiling_face: